packages/logger/src/Logger.ts

```diff
@@ -489,7 +490,13 @@ class Logger extends Utility implements LoggerInterface {
    * potentially yielding a different outcome.
    */
   public refreshSampleRateCalculation(): void {
-    this.setInitialSampleRate(this.powertoolsLogData.sampleRateValue);
+    /**
+     * we don't want to refresh the sample rate for cold starts
+     * because it might lead to double sampling, one in constructor and one here
```

Comment:
I would move this comment in the main docstring above explaining the same thing: when called during a cold start it results in a no-op.

This will be useful for those customers who use the manual method and call this directly.

Reply:
Agree!